docs/src/prototypes/chatMessageWithPopover/ChatMessageWithPopover.tsx

```diff
@@ -0,0 +1,102 @@
+import { Chat, Provider } from '@stardust-ui/react'
+
+import * as React from 'react'
+import Popover from './Popover'
+
+const janeAvatar = {
+  image: 'public/images/avatar/small/ade.jpg',
+  status: { color: 'green', icon: 'check' },
+}
+
+class ChatMessageWithPopover extends React.Component {
+  state = {
+    focused: false,
+  }
+
+  changeFocusState = (isFocused: boolean) => {
+    this.state.focused !== isFocused && this.setState({ focused: isFocused })
+  }
+
+  handleFocus = () => {
+    this.changeFocusState(true)
+  }
+
+  handleBlur = e => {
+    const shouldPreserveFocusState = e.currentTarget.contains(e.relatedTarget)
+    this.changeFocusState(shouldPreserveFocusState)
+  }
+
+  render() {
+    return (
+      <Chat.Message
+        author="Jane Doe"
+        timestamp="Yesterday, 10:15 PM"
+        content={{
+          content: (
+            <div>
+              <Popover className="actions" />
+              <a href="/">Link</a> Hover me to see the actions <a href="/">Some Link</a>
+            </div>
+          ),
+        }}
+        avatar={janeAvatar}
+        onFocus={this.handleFocus}
+        onBlur={this.handleBlur}
+        className={this.state.focused ? 'focused' : undefined}
```

Comment:
Please add `className` property to the API of this component. With the current implementation we are not respecting the user's input for this property. More over, the user currently has no way of styling it at all, except by using the `style` prop, which is not something we want to advertise! That way you can refactor the the className value in the following way: `className={cx(this.props.className, this.state.focused ? 'focused' : '')}`. More over, this is also a good example for using the createComponent paradigm, as in that way you would allow the user to use the ChatMessageWithProvider exactly the same way as they would use the ChatMessage.